### DISCUSSION TITLE
Is there a way to exceed the 5000 item limit when requesting documents?

### DISCUSSION THREAD

**Discussion by stuartcporter** [pnp/pnpcore#919] [Q&A] [answered]
hi,
we have sharepoint libraries with alot of files - and need to process them , using pnp.powershell you could retrieve using a pagesize  of 4000 at a time.

is the same feature available using pnp core? 

thx
stuart

**Comment by stuartcporter** [comment, marked as answer]
https://pnp.github.io/pnpcore/using-the-sdk/listitems-intro.html#using-paging-with-loaditemsbycamlquery following this 

I get that QueryProperties - is not an option? neither is AsRequested()
I'm using .net core 6.0 - which I believe is supported.

thx

**Reply by stuartcporter** [reply]
ok - I was missing a using - fair enough 
however after adding that and running with paging as per the example above
I get "The attempted operation is prohibited because it exceeds the list view threshold." surely the whole idea of paging is to circumvent this limit?

thx

**Reply by jansenbe** [reply]
@stuartcporter : can you verify if this solution works for you: https://github.com/pnp/pnpcore/discussions/586

**Reply by stuartcporter** [reply]
@jansenbe  Yes - I can confirm that this approach is working - thanks

**Comment by DaleyKD** [comment]
I know this doesn't help at this exact moment, but we use the data stream
and never have LVT issues.

**Reply by jansenbe** [reply]
@stuartcporter : here's an example of that: https://pnp.github.io/pnpcore/using-the-sdk/listitems-intro.html#using-paging-with-listdataasstream
